smdebug/core/state_store.py

```diff
@@ -79,7 +79,9 @@ def is_checkpoint_updated(self):
         """
         if self._checkpoint_dir is not None:
             checkpoint_timestamp = max(
-                os.path.getmtime(child) for child, _, _ in os.walk(self._checkpoint_dir)
+                os.path.getmtime(child)
+                for child, _, _ in os.walk(self._checkpoint_dir)
+                if child != os.path.dirname(self._states_file)
```

Comment:
why dirname? You need to check for exact filename and filter out meradata.json 
Also write a test case for this.